From: Helge Deller <deller@gmx.de>
Date: Sat, 22 Mar 2014 11:04:01 -0400
Subject: FTBFS hppa: correcting code inside "JS_STACK_GROWTH_DIRECTION > 0"

hppa is the only platform with growing stack upwards. Because of this
the code inside the

 #if JS_STACK_GROWTH_DIRECTION > 0
   ...
 #endif

was never touched by the compiler before and the wrong code inside this
statements could treated some build errors.

Closes: #741245
---
 mozilla/js/src/gc/RootMarking.cpp | 2 +-
 mozilla/js/src/jsapi.cpp          | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/mozilla/js/src/gc/RootMarking.cpp b/mozilla/js/src/gc/RootMarking.cpp
index 88574f3..25d94f3 100644
--- a/mozilla/js/src/gc/RootMarking.cpp
+++ b/mozilla/js/src/gc/RootMarking.cpp
@@ -320,7 +320,7 @@ MarkConservativeStackRoots(JSTracer *trc, bool useSavedRoots)
 
     uintptr_t *stackMin, *stackEnd;
 #if JS_STACK_GROWTH_DIRECTION > 0
-    stackMin = rt->nativeStackBase;
+    stackMin = reinterpret_cast<uintptr_t *>(rt->nativeStackBase);
     stackEnd = cgcd->nativeStackTop;
 #else
     stackMin = cgcd->nativeStackTop + 1;
diff --git a/mozilla/js/src/jsapi.cpp b/mozilla/js/src/jsapi.cpp
index 68b6afa..3c79597 100644
--- a/mozilla/js/src/jsapi.cpp
+++ b/mozilla/js/src/jsapi.cpp
@@ -864,9 +864,9 @@ JSRuntime::JSRuntime(JSUseHelperThreads useHelperThreads)
     PodZero(&debugHooks);
     PodZero(&atomState);
 
-#if JS_STACK_GROWTH_DIRECTION > 0
-    nativeStackLimit = UINTPTR_MAX;
-#endif
+//#if JS_STACK_GROWTH_DIRECTION > 0
+//    nativeStackLimit = UINTPTR_MAX;
+//#endif
 }
 
 bool
